src/main/java/Duke.java

```diff
@@ -1,10 +1,163 @@
+        int indexOfTo = taskDescription.indexOf("/to");
+        String description = taskDescription.substring(0, indexOfFrom);
+        String from = taskDescription.substring(
+                indexOfFrom + "/from ".length(),
+                indexOfTo
+        );
+        String to = taskDescription.substring(indexOfTo + "/to ".length());
+        Event newEvent = new Event(description, from, to);
+        tasks[counter] = newEvent;
+        counter += 1;
+        System.out.printf(
+                Message.TASK_ADDED.message,
+                newEvent.getDescriptionText(),
+                counter
+        );
+    }
+
+    /**
+     * Toggles the task as mark or un-mark based on the user
+     * request. Combined the two tasks together to abide by
+     * Single Responsibility Principle and Don't Repeat
+     * Yourself
+     *
+     * @param isMark Whether the user wants to mark the task
+     * @param index  The index of the task to be marked
+     */
+    private static void toggleMark(boolean isMark, int index) {
+        tasks[index].setDone(isMark);
+        System.out.printf(
+                "%s\n%s\n",
+                isMark ? Message.MARKED.message : Message.UNMARKED.message,
+                tasks[index].getDescriptionText()
+        );
+    }
+
+    /**
+     * Terminates the programme if the /bye command
+     * is given
+     */
+    private static void terminate() {
+        isRunning = false;
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner in = new Scanner(System.in);
+        System.out.println(Message.WELCOME.message);
+
+        while (isRunning) {
+            if (!in.hasNextLine()) {
+                break;
+            }
+
+            String line = in.nextLine().trim();
+            String[] words = line.split(" ", 2);
+            String command = words[0];
+            String taskDescription;
+
+            switch (command) {
+            case "list":
+                printTasks();
+                break;
+            case "deadline":
+                taskDescription = words[1];
+                addDeadlineTask(taskDescription);
+                break;
+            case "todo":
+                taskDescription = words[1];
+                addTodoTask(taskDescription);
+                break;
+            case "event":
+                taskDescription = words[1];
+                addEventTask(taskDescription);
+                break;
+            case "mark":
+            case "unmark":
+                int index = Integer.parseInt(words[1]) - 1;
+                boolean isMark = words[0].equals("mark");
+                toggleMark(isMark, index);
+                break;
+            case "bye":
```

Comment:
Perhaps it would be better to encapsulate the different command words into an enum so as to use LIST instead of "list" for the cases